src/commonMain/kotlin/io/github/petertrr/diffutils/text/DiffRowGenerator.kt

```diff
@@ -299,235 +273,199 @@ public class DiffRowGenerator(
     private fun generateInlineDiffs(delta: Delta<String>): List<DiffRow> {
         val orig = normalizeLines(delta.source.lines)
         val rev = normalizeLines(delta.target.lines)
-        val joinedOrig: String = orig.joinToString("\n")
-        val joinedRev: String = rev.joinToString("\n")
-        val origList = inlineDiffSplitter.invoke(joinedOrig).toMutableList()
-        val revList = inlineDiffSplitter.invoke(joinedRev).toMutableList()
-        // copying of `origList` and `revList` is needed because otherwise `wrapInTag` results in ConcurrentModificationException
-        val inlineDeltas: MutableList<Delta<String>> = diff(origList.toList(), revList.toList(), equalizer)
-            .deltas
-            .asReversed()
-            .toMutableList()
+        val joinedOrig = orig.joinToString("\n")
+        val joinedRev = rev.joinToString("\n")
+        val origList = inlineDiffSplitter.split(joinedOrig)
+        val revList = inlineDiffSplitter.split(joinedRev)
+
+        // Copying of `origList` and `revList` is needed because otherwise `wrapInTag` results in ConcurrentModificationException
```

Comment:
@petertrr I don't understand why this would throw a `ConcurrentModificationException`

Reply:
I'm not sure TBH; this part was copied from the original library.

Reply:
@petertrr no problem, I can remove it, it's not needed as we aren't operating on the lists concurrently.